pkg/sriov/sriov.go

```diff
@@ -169,14 +169,22 @@ func (s *sriovManager) ReleaseVF(conf *sriovtypes.NetConf, podifName string, cid
 			return fmt.Errorf("failed to rename link %s to host name %s: %q", podifName, conf.OrigVfState.HostIFName, err)
 		}
 
+		linkObj, err = s.nLink.LinkByName(conf.OrigVfState.HostIFName)
+		if err != nil {
+			return fmt.Errorf("failed to get netlink original device with name %s: %q", podifName, err)
+		}
+
 		// reset effective MAC address
 		if conf.MAC != "" {
 			hwaddr, err := net.ParseMAC(conf.OrigVfState.EffectiveMAC)
 			if err != nil {
 				return fmt.Errorf("failed to parse original effective MAC address %s: %v", conf.OrigVfState.EffectiveMAC, err)
 			}
 
-			if err = s.nLink.LinkSetHardwareAddr(linkObj, hwaddr); err != nil {
+			err = utils.Retry(5, 200*time.Millisecond, func() error {
```

Comment:
should we try to do things in reverse(ish) of setupVF ?

that is:
1. bring interface down
2. restore mac
3. set netdev orig name
4. move ifc back to host ns

WDYT ?

this way you may not need the extra call to LinkByName


In addition, can you add same comment for the Retry call as in SetupVF ?

Reply:
I change the solution a bit when you have time please take a look.

in general I prefer to leave the rename before because if for some reason, in any case, we failed to reset the mac address the container runtime will remove the network namespace and the VF will go back to the node with the pod nic name. 
how it's today it will always  have the right vf name even if for some driver issues we are not able to recover the right mac address.

Reply:
>in general I prefer to leave the rename before because if for some reason, in any case, we failed to reset the mac address the container runtime will remove the network namespace and the VF will go back to the node with the pod nic name.
how it's today it will always have the right vf name even if for some driver issues we are not able to recover the right mac address.

ok